File: java/spring-boot-rabbitmq/rabbitmq-domain/src/main/java/com/meridal/examples/springbootrabbitmq/domain/Song.java
package com.meridal.examples.springbootrabbitmq.domain;

import org.apache.commons.lang3.builder.ToStringBuilder;

import java.io.Serializable;
import java.util.Locale;

public class Song implements Serializable {

    private static final long serialVersionUID = 1L;

    private String title;
    private String duration;
    
    public String getTitle() {
        return this.title;
    }
    
    public String getDuration() {
        return this.duration;
    }
    
    public void setTitle(String title) {
        this.title = title;
    }
    
    public void setDuration(String duration) {
        this.duration = duration;
    }

    public void setDuration(int mins, int secs) {
        this.duration = this.formatTimePart(mins) + ":" + this.formatTimePart(secs);
    }

    private String formatTimePart(int timePart) {
        return String.format(Locale.UK, "%02d", timePart);
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this);
    }
}

File: kotlin/kotlin-domain/src/main/kotlin/com/meridal/examples/domain/Song.kt
package com.meridal.examples.domain

import java.io.Serializable
import java.util.*

class Song() : Serializable {

    var title: String? = null
    var duration: String? = null

    fun setDuration(mins: Int, secs: Int) {
        duration = formatTimePart(mins) + ":" + formatTimePart(secs)
    }

    private fun formatTimePart(timePart: Int): String {
        return String.format(Locale.UK, "%02d", timePart)
    }
}
